app/models/user.rb

```diff
@@ -0,0 +1,29 @@
+class User < ApplicationRecord
+  before_save { name.capitalize! }
+  
+  validates :name, presence: true, length: {within: 3..50}
+  validates_uniqueness_of :name
```

Comment:
This is cool, but realistically I am thinking name would not be unique, you'd be forcing someone with the same name as someone else to change their name. Maybe username, or better yet, you should add an email attribute!